redhat/overlays/log_server/Dockerfile

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright 2021 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.20.7@sha256:bc5f0b5e43282627279fe5262ae275fecb3d2eae3b33977a7fd200c7a760d6f1 AS builder
```

Comment:
This and the `deploy` image below should be using Red Hat images, as happens here: https://github.com/securesign/rekor/blob/e216f0058ab54f909ae141fda2b60189980d5fb9/redhat/patches/0001-dockerfile.patch#L10C7-L10C121